mono/metadata/marshal.c

```diff
@@ -8571,26 +8563,17 @@ mono_marshal_emit_managed_wrapper (MonoMethodBuilder *mb, MonoMethodSignature *i
 	 *   <interrupt check>
 	 *
 	 *   ret = method (...);
-	 * } catch (Exception e) {
-	 *   ex = mono_gchandle_new (e, false);
 	 * } finally {
```

Comment:
We need to remove the finally clause here, as there are 2 possible cases:
 - there is an exception (and we assume there is no `ftnptr_eh_callback`): we are going to switch GC unsafe -> safe here so we end up unwinding in GC safe mode which is not allowed.
 - there is no exception: it works properly

Reply:
You mean remove the clause but keep the detach call right ?